WebDriverAgentLib/Categories/XCUIApplication+FBHelpers.m

```diff
@@ -16,16 +16,59 @@
 #import "FBMathUtils.h"
 #import "FBXCodeCompatibility.h"
 #import "FBXPath.h"
+#import "FBXCTestDaemonsProxy.h"
+#import "FBXCAXClientProxy.h"
+#import "XCAccessibilityElement.h"
 #import "XCElementSnapshot+FBHelpers.h"
 #import "XCUIDevice+FBHelpers.h"
 #import "XCUIElement+FBIsVisible.h"
 #import "XCUIElement+FBUtilities.h"
 #import "XCUIElement+FBWebDriverAttributes.h"
+#import "XCTestManager_ManagerInterface-Protocol.h"
+#import "XCTestPrivateSymbols.h"
+#import "XCTRunnerDaemonSession.h"
 
 const static NSTimeInterval FBMinimumAppSwitchWait = 3.0;
 
 @implementation XCUIApplication (FBHelpers)
 
++ (XCAccessibilityElement *)fb_onScreenElement
+{
+  static CGPoint screenPoint;
+  static dispatch_once_t oncePoint;
+  dispatch_once(&oncePoint, ^{
+    CGSize screenSize = [UIScreen mainScreen].bounds.size;
+    // Consider the element, which is located close to the top left corner of the screen the on-screen one.
+    // FIXME: Improve this algorithm for split-screen automation
+    CGFloat pointDistance = MIN(screenSize.width, screenSize.height) * 0.2;
```

Comment:
Should we just hit the middle?

Reply:
How do we automate split screens btw? It looks like a weird case

Reply:
> Should we just hit the middle?

I just wanted to avoid touch point calculation every time we determine the active application (because of possible orientation change)

> How do we automate split screens btw? It looks like a weird case
 
Currently it is not implemented and we don't support such automation officially. However, a possible approach to that would be implementing of `/window` API, so the client could list, set and get apps on his own. Although, it will be a breaking change for the people having tests where they switch between multiple apps.